pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
+            import pathlib
+        except ImportError:
+            pytest.skip('need pathlib to run')
+        expected = self.read_csv(self.csv1, index_col=0, parse_dates=True)
+        for extension in '', '.gz', '.bz2':
+            path = pathlib.Path(self.csv1 + extension)
+            df = self.read_csv(
+                path, index_col=0, parse_dates=True, compression='infer')
```

Comment:
I would remove this test entirely. instead add a generalized compression test (iow for all extensions & path types) in ``test_common``, similar to how fspath is tested (you specificy all parameters then you may skip on some combinations).